### DISCUSSION TITLE
StartLinuxDrm no work in Avalonia 11.

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#11950] [Q&A] [answered]
Recently I am working on try run my program on Raspberry Pi with Raspbian Lite.

I followed this tutorial:
https://docs.avaloniaui.net/docs/next/guides/platforms/rpi/running-on-raspbian-lite-via-drm

I did all the steps one by one.

Whereas, in step 2.3 Prepare Program.cs , it said I should modify the code like this:
```
public static int Main(string[] args)
{
    var builder = BuildAvaloniaApp();
    if (args.Contains("--drm"))
    {
        SilenceConsole();
        return builder.StartLinuxDrm(args);
    }
    return builder.StartWithClassicDesktopLifetime(args);
}
```
I did what is said and Visual Studio reports an error:

> The call is ambiguous between the following methods or properties: 'LinuxFramebufferPlatformExtensions.StartLinuxDrm(AppBuilder, string[], string?, double, IInputBackend?)' and 'LinuxFramebufferPlatformExtensions.StartLinuxDrm(AppBuilder, string[], string?, bool, DrmOutputOptions?, IInputBackend?)'

My project was created by 11 Preview 5 a long time ago and continuously upgraded to 11 RC1. All the nuget package is on 11 RC1 also. Currently my program runs well on windows. What's wrong with it?

PS: I tried to create a brand new 11 RC1 project by VS and met the same problem. It seems there is something wrong with the tutorial.

**Comment by kyuranger** [comment, marked as answer]
The code should be like this:
```
[STAThread]
        public static int Main(string[] args)
        {
            var builder = BuildAvaloniaApp();
            double GetScaling()
            {
                var idx = Array.IndexOf(args, "--scaling");
                if (idx != 0 && args.Length > idx + 1 &&
                    double.TryParse(args[idx + 1], NumberStyles.Any, CultureInfo.InvariantCulture, out var scaling))
                    return scaling;
                return 1;
            }
            if (args.Contains("--drm"))
            {
                SilenceConsole();
                return builder.StartLinuxDrm(args: args, card: "/dev/dri/card1", scaling: GetScaling());
            }

            return builder.StartWithClassicDesktopLifetime(args);
        }
```

**Reply by TonWin618** [reply]
The "DefaultPropertyHelper" class is located in which namespace?

**Reply by kyuranger** [reply]
That's use to store some setting temporarily in my project and it is useless for this topic. Now I deleted this line of code in the answer.
